pkg/types/validation/installconfig.go

```diff
@@ -990,14 +992,28 @@ func validateIPProxy(proxy string, n *types.Networking, fldPath *field.Path) fie
 // for ssh keys on FIPS.
 func validateFIPSconfig(c *types.InstallConfig) field.ErrorList {
 	allErrs := field.ErrorList{}
-	sshParsedKey, _, _, _, err := ssh.ParseAuthorizedKey([]byte(c.SSHKey))
-	if err != nil {
-		allErrs = append(allErrs, field.Invalid(field.NewPath("sshKey"), c.SSHKey, fmt.Sprintf("Fatal error trying to parse configured public key: %s", err)))
-	} else {
-		sshKeyType := sshParsedKey.Type()
-		re := regexp.MustCompile(`^ecdsa-sha2-nistp\d{3}$|^ssh-rsa$`)
-		if !re.MatchString(sshKeyType) {
-			allErrs = append(allErrs, field.Invalid(field.NewPath("sshKey"), c.SSHKey, fmt.Sprintf("SSH key type %s unavailable when FIPS is enabled. Please use rsa or ecdsa.", sshKeyType)))
+	if c.SSHKey != "" {
+		sshParsedKey, _, _, _, err := ssh.ParseAuthorizedKey([]byte(c.SSHKey))
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("sshKey"), c.SSHKey, fmt.Sprintf("Fatal error trying to parse configured public key: %s", err)))
+		} else {
+			sshKeyType := sshParsedKey.Type()
+			re := regexp.MustCompile(`^ecdsa-sha2-nistp\d{3}$|^ssh-rsa$`)
+			if !re.MatchString(sshKeyType) {
+				allErrs = append(allErrs, field.Invalid(field.NewPath("sshKey"), c.SSHKey, fmt.Sprintf("SSH key type %s unavailable when FIPS is enabled. Please use rsa or ecdsa.", sshKeyType)))
+			}
+		}
+	}
+
+	if err := hostcrypt.VerifyHostTargetState(c.FIPS); err != nil {
+		if _, ok := os.LookupEnv("OPENSHIFT_INSTALL_SKIP_HOSTCRYPT_VALIDATION"); ok {
```

Comment:
Just a couple of comments/questions:

* If `OPENSHIFT_INSTALL_SKIP_HOSTCRYPT_VALIDATION` was set to `true`, shouldn't be skipped at all `hostcrypt.VerifyHostTargetState`?
* The `LookupEnv` doesn't check the value of `OPENSHIFT_INSTALL_SKIP_HOSTCRYPT_VALIDATION` (so the annotation could be set to `true` even though the var was set `false` - or something else)

Reply:
We still want to call VerifyHostTargetState unconditionally so that we can at least log a warning and record that this happened.
It's true that we're only checking for the existence of the environment variable here, not the value. I'm relaxed about that. It's consistent with e.g. `OPENSHIFT_INSTALL_PRESERVE_BOOTSTRAP`.

Reply:
> We still want to call VerifyHostTargetState unconditionally so that we can at least log a warning and record that this happened. 
Not a blocking point, but for the sake of clarity it may be useful for the end user then to mention in the warning log trace that the fips errors are ignored (since the env var is set)

>It's true that we're only checking for the existence of the environment variable here, not the value. I'm relaxed about that. It's consistent with e.g. `OPENSHIFT_INSTALL_PRESERVE_BOOTSTRAP`.
The `OPENSHIFT_INSTALL_PRESERVE_BOOTSTRAP` at least checks that the value is not empty to verify that it's set. In any case, not checking the (accepted) value IMHO introduces an ambiguous user interface that, once available, it will never change in future